src/pgrx_tests/common.rs

```diff
@@ -0,0 +1,307 @@
+use std::fs::File;
+use std::marker::PhantomData;
+use std::{collections::HashMap, fmt::Debug};
+
+use crate::type_compat::map::Map;
+
+use arrow::array::RecordBatch;
+use arrow_schema::SchemaRef;
+use parquet::arrow::ArrowWriter;
+use pgrx::{
+    datum::{Time, TimeWithTimeZone},
+    FromDatum, IntoDatum, Spi,
+};
+
+pub(crate) enum CopyOptionValue {
+    StringOption(String),
+    IntOption(i64),
+}
+
+pub(crate) fn comma_separated_copy_options(options: &HashMap<String, CopyOptionValue>) -> String {
+    let mut comma_sepated_options = String::new();
+
+    for (option_idx, (key, value)) in options.iter().enumerate() {
+        match value {
+            CopyOptionValue::StringOption(value) => {
+                comma_sepated_options.push_str(&format!("{} '{}'", key, value));
+            }
+            CopyOptionValue::IntOption(value) => {
+                comma_sepated_options.push_str(&format!("{} {}", key, value));
+            }
+        }
+
+        if option_idx < options.len() - 1 {
+            comma_sepated_options.push_str(", ");
+        }
+    }
+
+    comma_sepated_options
+}
+
+pub(crate) struct TestTable<T: IntoDatum + FromDatum> {
+    uri: String,
+    order_by_col: String,
+    copy_to_options: HashMap<String, CopyOptionValue>,
+    copy_from_options: HashMap<String, CopyOptionValue>,
+    _data: PhantomData<T>,
+}
+
+impl<T: IntoDatum + FromDatum> TestTable<T> {
+    pub(crate) fn new(typename: String) -> Self {
+        Spi::run("DROP TABLE IF EXISTS test_expected, test_result;").unwrap();
+
+        let create_table_command = format!("CREATE TABLE test_expected (a {});", &typename);
+        Spi::run(create_table_command.as_str()).unwrap();
+
+        let create_table_command = format!("CREATE TABLE test_result (a {});", &typename);
+        Spi::run(create_table_command.as_str()).unwrap();
+
+        let mut copy_to_options = HashMap::new();
+        copy_to_options.insert(
+            "format".to_string(),
+            CopyOptionValue::StringOption("parquet".to_string()),
+        );
+
+        let mut copy_from_options = HashMap::new();
+        copy_from_options.insert(
+            "format".to_string(),
+            CopyOptionValue::StringOption("parquet".to_string()),
+        );
+
+        let uri = "/tmp/test.parquet".to_string();
```

Comment:
maybe should use more obscure file names. A user/developer might have an unrelated file named this way and get very confused